src/pudl_rmi/connect_deprish_to_ferc1.py

```diff
@@ -66,762 +191,380 @@ def execute(plant_parts_eia, deprish_eia, ferc1_to_eia, clobber=False):
 
-        Store instance of InputsCompiler so we can use the prepared
-        dataframes.
+    @validator('treatment_type')
+    def check_treatments(cls, value, values):   # noqa: N805
+        """Check treatments that need additional info."""
+        if value == 'scale' and not values.get('allocator_cols'):
+            raise AssertionError(
+                "Scale column treatment needs allocator_cols")
+        if value == 'wtavg' and not values.get('wtavg_col'):
+            raise AssertionError(
+                "Weighted Average column treatment needs wtavg_col")
+        return value
 
-        Args:
-            inputs (object): an instance of InputsCompiler
 
-        """
-        self.inputs = inputs
-        inputs.prep_inputs()
+class PlantPartScaler(BaseModel):
+    """
+    Scale a table to a plant-part.
 
-    def check_all_candidate_matches(self, candidate_matches_all):
-        """
-        Check the candidate matches between depreciation data and ferc1.
-
-        This method explores the generation of matches between depreciation
-        and ferc1 records. We want to know how many depreciation records aren't
-        associated with any ferc1 record. We want to know if there are any
-        plant ids that show up in the depreciation data and aren't mapped to
-        ferc records but do show up in the ferc data somewhere. At a high
-        level, we want a gut check of whether or not connects_all_deprish_ferc1
-        was connected properly.
-        """
-        # there was a merge iindicator here and left df was the depreciation
-        # data
-        connected_plant_ids = candidate_matches_all[
-            candidate_matches_all._merge == 'both'].plant_id_pudl.unique()
-        # how many plant_id_pudl's didn't get a corresponding FERC1 record
-        not_in_ferc1_plant_ids = (
-            candidate_matches_all[candidate_matches_all._merge == 'left_only']
-            .plant_id_pudl.unique()
-        )
-        # these are a subset of the not_in_ferc1_plant_ids that show up in the
-        # steam table
-        missing_plant_ids = (self.inputs.connects_ferc1_eia[
-            self.inputs.connects_ferc1_eia.plant_id_pudl.isin(
-                not_in_ferc1_plant_ids)].plant_id_pudl.unique())
-        logger.info(f"Matched plants:    {len(connected_plant_ids)}")
-        logger.info(f"Not connected:       {len(not_in_ferc1_plant_ids)}")
-        logger.info(f"Missing connections: {len(missing_plant_ids)}")
-        # Investigation of 3 missing connections [1204, 1147, 1223] determined
-        # that these plants are missing from ferc because there was no
-        # reporting for the specific years in question
-        if len(missing_plant_ids) > 3:
-            warnings.warn(
-                f'There are {len(missing_plant_ids)} missing plant records.')
-
-    def get_candidate_matches(self):
-        """
-        Prepare all candidate matches between depreciation and ferc1 steam.
+    Args:
```

Comment:
Docstring?  Can `plant_part` only have a single value, and this is just a placeholder for being able to scale to other granularities?

Reply:
yes.... i've considered removing this from the args but I want it to be very obvious that we are scaling to plant_gen. it will fail if you try something differnt